### DISCUSSION TITLE
Start OpenThread Border Router

### DISCUSSION THREAD

**Discussion by chuckrui** [openthread/openthread#9497] [Q&A]
**Listed below are my debugging steps.**

(1)Install Border Router by docker
`sudo systemctl start docker`

(2)Create IPV6 network for Border Router on docker 
`sudo docker network create --ipv6 --subnet fd11:db8:1::/64 -o com.docker.network.bridge.name=otbr0 otbr` 

(3)Download a compatible version of the OpenThread border Router docker image
`sudo docker pull nrfconnect/otbr:a892bf7`

(4)Ensure that packets sent to devices in the Thread network are routed through the OpenThread Border Router container in Docker
`sudo ip -6 route add fd11:22::/64 dev otbr0 via fd11:db8:1::2     `

(5)Run modprobe to load the iptables kernel module
`sudo modprobe ip6table_filter	`

(6)Start the OpenThread border router container
`sudo docker run -it --rm --privileged --name otbr --network otbr -p 8080:80 \
--sysctl "net.ipv6.conf.all.disable_ipv6=0 net.ipv4.conf.all.forwarding=1 net.ipv6.conf.all.forwarding=1" \
--volume /dev/ttyACM0:/dev/radio nrfconnect/otbr:a892bf7 --radio-url spinel+hdlc+uart:///dev/radio?uart-baudrate=1000000`

**log run by the last command.**
I don't know why sometimes it succeeds and sometimes it fails. In the web side, the network From thread will also fail. I believe it is because of the error of this last command, because if it is normal, after clicking the From button in the web side, it will be successful.

[otbr_log.txt](https://github.com/openthread/openthread/files/12842480/otbr_log.txt)

**Comment by chuckrui** [comment]
Sometimes there will be such a log, which also fails to open OTBR, can you tell me the reason?
[log.txt](https://github.com/openthread/openthread/files/12842547/log.txt)

**Comment by jwhui** [comment]
It appears that `otbr-agent` is not able to communicate with the RCP. It could be that the RCP is not configured properly or the Radio URL is not configure properly.

**Comment by chuckrui** [comment]
> It appears that `otbr-agent` is not able to communicate with the RCP. It could be that the RCP is not configured properly or the Radio URL is not configure properly.

@jwhui  Thank you for your answer.
However, the strange thing is that with the same RCP (52840dongle) and the same steps, I successfully opened the web interface and carried out From the Raspberry PI, successfully built the Thread network, and successfully used the Chip-tool to distribute the network. However, when I tried to use the UBuntu20.04 physical machine, the error in the above appended log was displayed. What are the differences between Raspberry PI and Ubuntu20.04 that will affect the build of the Thread network?

**Reply by abtink** [reply]
@chuckrui can you please share which commit of OT are you using?

**Reply by chuckrui** [reply]
@abtink  Thanks for you help.
Specific version may not be clear to me, I am using Nordic52840, then burn replication 2.3.0 version of the directory path the `nrf/samples/openthread/coprocessor` project, I did not directly clone OT project of warehouse, But I see [this](https://github.com/openthread/ot-nrf528xx/blob/main/src/nrf52840/README.md) is related with me? 
Now I want to create a Thread network stably, and then I can obtain dataset active and debug the thread network, but the same hardware (52840dongle) and the same steps can run successfully on the Raspberry PI, but fail on the Ubuntu physical machine. Sometimes it's "Wpan service error:13", sometimes it's "Wpan service error:8"

**Comment by chuckrui** [comment]
**Is it related to the network card name? The ifconfig log on the Raspberry PI is as follows.**
docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:32:ac:79:d7  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether dc:a6:32:be:ba:e8  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 873  bytes 85262 (85.2 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 873  bytes 85262 (85.2 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

otbr0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.18.0.1  netmask 255.255.0.0  broadcast 172.18.255.255
        inet6 fe80::42:7cff:febe:e5bd  prefixlen 64  scopeid 0x20<link>
        inet6 fe80::1  prefixlen 64  scopeid 0x20<link>
        inet6 fd11:db8:1::1  prefixlen 64  scopeid 0x0<global>
        ether 02:42:7c:be:e5:bd  txqueuelen 0  (Ethernet)
        RX packets 1529  bytes 1927235 (1.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 648  bytes 51604 (51.6 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth2c1f8ff: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::c086:ceff:fed2:38ab  prefixlen 64  scopeid 0x20<link>
        ether c2:86:ce:d2:38:ab  txqueuelen 0  (Ethernet)
        RX packets 1529  bytes 1948641 (1.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 689  bytes 58949 (58.9 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.3.39  netmask 255.255.255.0  broadcast 192.168.3.255
        ether dc:a6:32:be:ba:e9  txqueuelen 1000  (Ethernet)
        RX packets 61362  bytes 80966026 (80.9 MB)
        RX errors 0  dropped 5  overruns 0  frame 0
        TX packets 31043  bytes 4569921 (4.5 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        
**Ubuntu20.04：**
`docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:0d:e6:87:d1  txqueuelen 0  (以太网)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (本地环回)
        RX packets 14393  bytes 1425243 (1.4 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 14393  bytes 1425243 (1.4 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

otbr0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.18.0.1  netmask 255.255.0.0  broadcast 172.18.255.255
        inet6 fd11:db8:1::1  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::42:eff:fea8:10c1  prefixlen 64  scopeid 0x20<link>
        inet6 fe80::1  prefixlen 64  scopeid 0x20<link>
        ether 02:42:0e:a8:10:c1  txqueuelen 0  (以太网)
        RX packets 7342  bytes 7788301 (7.7 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 3554  bytes 263322 (263.3 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlo1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.3.77  netmask 255.255.255.0  broadcast 192.168.3.255
        inet6 fe80::fec3:ae11:96f0:a549  prefixlen 64  scopeid 0x20<link>
        ether f4:6a:dd:6c:d6:0f  txqueuelen 1000  (以太网)
        RX packets 0  bytes 32445582 (32.4 MB)
        RX errors 0  dropped 6  overruns 0  frame 0
        TX packets 4519  bytes 1139164 (1.1 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
